ibis/impala/compiler.py

```diff
@@ -125,6 +126,9 @@ def _cast(translator, expr):
 
     if isinstance(arg, ir.CategoryValue) and target_type == 'int32':
         return arg_formatted
+    if (isinstance(arg, (ir.TimeValue, ir.DateValue, ir.TimestampValue)) and
+            target_type == 'int64'):  # noqa E129
+        return '1e6 * unix_timestamp({})'.format(arg_formatted)
```

Comment:
You're multiplying by `1e6` which will cause the result type to be a `double`, even though the type we're casting to is `int64`. Is there any particular reason why you're doing that?

Reply:
Also, if you make this change you need to add a test. If you do not make this change then you don't